src/logic/relation.lean

```diff
@@ -390,6 +400,18 @@ lemma trans_gen.closed {p : α → α → Prop} :
   (∀ a b, r a b → trans_gen p a b) → trans_gen r a b → trans_gen p a b :=
 trans_gen.lift' id
 
+lemma trans_gen.mono {p : α → α → Prop} :
+  (∀ a b, r a b → p a b) → trans_gen r a b → trans_gen p a b :=
+trans_gen.lift id
+
+lemma trans_gen_swap : trans_gen (swap r) a b ↔ trans_gen r b a :=
```

Comment:
It might be worth having `trans_gen.swap : trans_gen r a b → trans_gen (swap r) b a` for a nice dot notatable one way implication, but it's not strictly necessary.